core/metrics-core-service/src/main/java/org/hawkular/metrics/core/service/MetricsServiceImpl.java

```diff
@@ -1007,23 +1006,17 @@ public void shutdown() {
     }
 
     @Override
+    @SuppressWarnings("unchecked")
     public <T> Observable<Void> deleteMetric(MetricId<T> id) {
-        Observable<Void> result = dataAccess.getMetricTags(id)
-                .map(row -> row.getMap(0, String.class, String.class))
-                .defaultIfEmpty(new HashMap<>())
-                .flatMap(map -> dataAccess.deleteFromMetricsTagsIndex(id, map))
-                .toList()
-                .flatMap(r -> dataAccess.deleteMetricFromMetricsIndex(id))
-                .flatMap(r -> null);
-
         //NOTE: compressed data is not deleted due to the using TWCS compaction strategy
         //      for the compressed data table.
-        result.mergeWith(dataAccess.deleteMetricData(id).flatMap(r -> null));
-
-        result.mergeWith(dataAccess.deleteMetricFromRetentionIndex(id).flatMap(r -> null));
-        result.mergeWith(dataAccess.deleteFromMetricExpirationIndex(id).flatMap(r -> null));
 
-        return result;
+        return getMetricTags(id)
```

Comment:
I think flatMap is still called because the observable returned from getMetricTags still emits an empty map.